native/packages/react-native-bpk-component-animate-height/src/BpkAnimateHeight.js

```diff
@@ -16,126 +16,122 @@
 
 const COLLAPSED_HEIGHT = 0.01;
 
+const STYLES = StyleSheet.create({
+  view: {
+    overflow: 'hidden',
+    borderRadius: 0, // overflow hidden hack for Android
+  },
+});
+
 class BpkAnimateHeight extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      expanded: this.props.expanded,
-      height: new Animated.Value(COLLAPSED_HEIGHT),
-      inTree: this.props.expanded,
-      expandAnimationInProgress: false,
-    };
-
-    // overflow: hidden is not properly supported on Android,
-    // which causes animation to look shoddy.
-    // See https://facebook.github.io/react-native/releases/0.49/docs/layout-props.html#overflow
-    this.animationDuration =
-      Platform.OS === 'ios' ? props.animationDuration : 0;
+    this.innerViewRef = null;
 
-    this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
-    this.setExpandedHeight = this.setExpandedHeight.bind(this);
-    this.onHeightAnimationComplete = this.onHeightAnimationComplete.bind(this);
-    this.performAnimation = this.performAnimation.bind(this);
+    this.height = this.props.expanded
+      ? null
+      : new Animated.Value(COLLAPSED_HEIGHT);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.expanded !== this.state.expanded) {
-      if (this.state.expanded) {
-        this.setState({ expanded: false });
-        this.performAnimation(COLLAPSED_HEIGHT, this.props.collapseDelay);
-      } else {
-        this.setState({ inTree: true, expandAnimationInProgress: true });
-      }
+  componentDidMount() {
+    if (this.height) {
+      return;
     }
-  }
 
-  onHeightAnimationComplete() {
-    if (typeof this.props.onAnimationComplete === 'function') {
-      this.props.onAnimationComplete();
-    }
-    // eslint-disable-next-line no-underscore-dangle
-    if (this.state.height._value - 0.01 < 0.00001) {
-      this.setState({ inTree: false });
-    }
-  }
+    const captureHeight = (x, y, width, height) => {
+      this.height = new Animated.Value(height);
+    };
 
-  setExpandedHeight(event) {
-    const { height } = event.nativeEvent.layout;
-    if (height === 0) {
-      return;
-    }
-    if (!this.state.expandAnimationInProgress) {
-      this.setState({
-        height: new Animated.Value(
-          this.state.expanded ? height : COLLAPSED_HEIGHT,
-        ),
-      });
-    } else {
-      this.setState({ expanded: true, expandAnimationInProgress: false });
-      this.performAnimation(height, this.props.expandDelay);
-    }
+    this.measure(captureHeight);
   }
 
-  performAnimation(height, delay) {
-    Animated.timing(this.state.height, {
-      toValue: height,
-      duration: this.animationDuration,
-      delay,
-    }).start(this.onHeightAnimationComplete);
+  componentDidUpdate() {
+    this.measure(this.animate);
   }
 
-  render() {
+  measure = callback =>
+    requestAnimationFrame(() => this.innerViewRef.measure(callback));
```

Comment:
Use react-natives "direct manipulation" measure api to get the height for the inner view - see https://facebook.github.io/react-native/docs/direct-manipulation.html#measurecallback